pkg/api/v1beta3/types.go

```diff
@@ -827,6 +827,10 @@ type PodSpec struct {
 	// used must be specified.
 	// Optional: Default to false.
 	HostNetwork bool `json:"hostNetwork,omitempty" description:"host networking requested for this pod"`
+	// ImagePullSecrets is an optional list of references to secrets in the same namespace to use for pulling any of the images used by this PodSpec.
+	// If specified, these secrets will be passed to individual puller implementations for them to use.  For example,
+	// in the case of docker, only DockerConfig type secrets are honored.
+	ImagePullSecrets []ObjectReference `json:"imagePullSecrets,omitempty" description:"list of references to secrets in the same namespace available for pulling the container images"`
```

Comment:
You need structured objects to use patchStrategy/patchMergeKey if you want to be able to remove one of the items via patch. The only patch operation you can do with a slice of scalars is merge/dedup everything in both slices.